src/index.js

```diff
@@ -9,9 +9,22 @@ const { allRoles, roles } = require("./roles.js")
 const aliases = require("./aliases.js")
 const getRole = (rolename) => {
   rolename = rolename.toLowerCase()
-  rolename = rolename.replace(/-/g, " ")
+  rolename = rolename.replace(/-/g, " ").replace(/_/g, " ")
   rolename = aliases[rolename] ? aliases[rolename] : rolename
   let foundRole = roles.find(role => role.name.toLowerCase() === rolename) || roles.find(role => role.name.toLowerCase().startsWith(rolename)) || roles.find(role => role.name.toLowerCase().endsWith(rolename)) || roles.find(role => role.name.toLowerCase().includes(rolename))
+  if (!foundRole) {
+    let theRole = rolename[0]
+    let lastFoundRole = null
+    for (let i = 1 ; i < rolename.length ; i++) {
+      let check = roles.find(role => role.name.toLowerCase() === theRole) || roles.find(role => role.name.toLowerCase().startsWith(theRole)) || roles.find(role => role.name.toLowerCase().endsWith(theRole)) || roles.find(role => role.name.toLowerCase().includes(theRole))
+      if (!check) {
+        foundRole = lastFoundRole
+        break;
+      }
```

Comment:
You could also make the function recursive

Reply:
Wdym? I added this for people who wrote the name wrongly like villagef instead of villager for example.

Reply:
Yeah ik. but instead of doing all this code just let the function call itself. Less code

Reply:
Yea i just got what you said, mb